### PR TITLE
support format priorities

### PR DESCRIPTION
**Pull request by lsmith77**
supersedes https://github.com/FriendsOfSymfony/FOSRestBundle/pull/1153
fixes https://github.com/FriendsOfSymfony/FOSRestBundle/issues/1151

### CODE REVIEW

**Comment by Ener-Getick** [comment]
+1, the only problem remaining is that the aliases are not configured by default. 
I think we can let this bc break.

**Comment by lsmith77** [comment]
yeah. i will add a note in the docs about this change

File: Negotiation/FormatNegotiator.php [hunk 1]
```diff
@@ -103,4 +113,47 @@ public function getBest($header, array $priorities = [])
             }
         }
     }
+
+    /**
+     * @param array $values
+     *
+     * @return array
+     */
+    protected function sanitize(array $values)
```
Comment (xabbuh) [inline Negotiation/FormatNegotiator.php]: Shouldn't these methods be private to not expose additionally extension points?

File: Negotiation/FormatNegotiator.php [hunk 2]
```diff
@@ -103,4 +113,47 @@ public function getBest($header, array $priorities = [])
...
+
+    /**
+     * Transform the format (json, html, ...) to their mimeType form (application/json, text/html, ...).
+     *
+     * @param Request $request
+     * @param array   $priorities
+     *
+     * @return array formated priorities
+     */
+    protected function normalizePriorities(Request $request, $priorities)
```
Comment (xabbuh) [inline Negotiation/FormatNegotiator.php]: What about adding an `array` type hint for `$priorities`?

Comment (lsmith77) [inline Negotiation/FormatNegotiator.php]: +1

File: Negotiation/FormatNegotiator.php (hunk 1 above)
Comment (lsmith77) [inline Negotiation/FormatNegotiator.php]: +1

Comment (xabbuh) [inline Negotiation/FormatNegotiator.php]: see #1162

**Pull request merged**